### ISSUE TITLE
Bigger warnings on website or docs regarding private info?

### ISSUE THREAD

**Issue by jzf2101** [jupyterhub/binderhub#613]
We got a lot of questions regarding putting private information eg private repos on binder.  we should probably have more documentation on this on the website and the docs. @betatim

**Comment by tgeorgeux** [comment]
@jzf2101 Which website are we talking about? I haven't really delved into binder-land at all yet. Are you talking about for mybinder.org?

**Comment by jzf2101** [comment]
Yeah on the website.  How do you think we should convey that info design wise to make it clear to people?

**Comment by tgeorgeux** [comment]
(From the website instructions, I would add the bracketed copy below)
Enter your repository information
Provide in the above form a URL or a GitHub repository that contains Jupyter notebooks, as well as a branch, tag, or commit hash. Launch will build your Binder repository. If you specify a path to a notebook file, the notebook will be opened in your browser after building. **[Binder is a free, public service that is open to the world, and you should never share sensitive or personal information within a Binder repository. This includes passwords, data that shouldn’t be public, API keys, etc.]**

I would also consider giving a contextual warning when a file is uploaded, but we can look more at that when we have some time to do a little dive into the context.

**Comment by choldgraf** [comment]
I agree this is definitely important. We're gonna need to do an update of the main binder page in general, so we should add some kind of language to this effect when we do so.

I'm hesitant to add all of the text that @tgeorgeux suggests to the main page (though I agree it's good content and we want it up somewhere). The main Binder landing page needs to be as simple and word-free as possible.

**Comment by jzf2101** [comment]
I think we could do something along those lines - like a contextual warning when people begin to enter a URL into the form?

**Comment by tgeorgeux** [comment]
Would it be fair to say the desire message is: "Do not upload sensitive or personal information!" or some iteration of that?

**Comment by jzf2101** [comment]
yes i think so

**Comment by tgeorgeux** [comment]
You can have that copy appear directly above above the URL window in red regular weight font (more intrusive), or you could put that copy inside the url area when it's not been filled out yet (less intrusive).

**Comment by choldgraf** [comment]
another thought is to add this not to the main `mybinder.org` landing page, but instead to the "loading" page that people see when they click a binder link. Those cases are more likely to be seen by users who might be "vulnerable" to this.

e.g.:

![image](https://user-images.githubusercontent.com/1839645/43483968-fab90920-94c1-11e8-9abe-6d013d4e30e6.png)

**Comment by jzf2101** [comment]
But wouldn't people be pushing vulnerable info to the repo itself? I was thinking it could also be in the form itself?

**Comment by choldgraf** [comment]
ohhh interesting, I hadn't thought of that. Could you give an example of what you're talking about? Do you imagine something in `urlpath` or something?

**Comment by jzf2101** [comment]
I was thinking we could put a warning in the repo2docker form to tell them not to build images with personal content

**Comment by choldgraf** [comment]
ahh - from my perspective, I'm less worried about people building images with personal content because they'll already have everything online in GitHub. I'm more worried about built images **asking** people for personal content after people click on a Binder link :-/

**Comment by jzf2101** [comment]
both are true

**Comment by betatim** [comment]
In addition to permanent changes that are discussed above should we create a specific page (or use the FAQ entry?) that explains the reasoning behind this? Could also mention other forms of data that you should not use (cc @ctb).

Once we have this, what do you think of having a Binder-and-your-secrets-awareness week? Tweets, some kind of "fork this on github" like ribbon in the corner of mybinder.org pages, short jupyter blog post?

**Comment by choldgraf** [comment]
that's a cool idea :-)
